### PR TITLE
Create test substreanm

### PR DESCRIPTION
**Pull request by harshrjjpt**
- Created a custom substream to get ethereum network Block Data and Transaction data

### CODE REVIEW

**Comment by realchoubey** [comment]
Remove:
- All auto-generated files.
- Build folder, as it has duplicate files.

File: network_substream/Cargo.lock
Comment (realchoubey) [inline network_substream/Cargo.lock:1]: Do we need this file?

File: network_substream/proto/block_meta.proto [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+// ## Block data ##
+message BlockMeta {
+  string hash = 1;
+  uint64 number = 2;
+  string parent_hash = 3;
+  string timestamp = 4;
+}
+message BlockHeader {
```
Comment (realchoubey) [inline network_substream/proto/block_meta.proto:10]: Add extra between message.

File: network_substream/proto/block_meta.proto [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  uint64 number = 6;
+  uint64 gasLimit = 7;
+  uint64 gasUsed = 8;
+  int64 timestamp = 9;
+  uint64 size = 10;
+  string difficulty = 11; 
+  string totalDifficulty = 12;
+}
+
+
```
Comment (realchoubey) [inline network_substream/proto/block_meta.proto]: 1 line between message block would be fine.

File: network_substream/proto/block_meta.proto [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  int32 gas_cost = 1;
+  string change_type = 2;
+}
+
+message AccountCreation {
+  bytes address = 1;
+  bytes balance = 2;
+  uint64 nonce = 3;
+  repeated StorageChange initial_storage = 4;
+}
```
Comment (realchoubey) [inline network_substream/proto/block_meta.proto]: Add new line at end.

File: network_substream/proto/block_meta.proto
Comment (realchoubey) [inline network_substream/proto/block_meta.proto:1]: Are we using all messages?

File: network_substream/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  number: BigInt!
+  gasLimit: BigInt!
+  gasUsed: BigInt!
+  timestamp: BigInt!
+  size: BigInt!
+  difficulty: String!
+  totalDifficulty: String!
+}
+
+# type BlockHeader @entity {
```
Comment (realchoubey) [inline network_substream/schema.graphql]: Remove unused entities

File: network_substream/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,188 @@
+#change it to TranactionDetails
```
Comment (realchoubey) [inline network_substream/schema.graphql]: These comment not required.

File: network_substream/src/eth.rs
Comment (realchoubey) [inline network_substream/src/eth.rs:1]: Are we using this file?

File: network_substream/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+use pb::acme::{Transaction, TransactionList, BlockHeader};
+use substreams_ethereum::pb::eth::v2 as eth;
+use substreams_ethereum::pb::eth::v2::TransactionTraceStatus;
+use substreams::store::{StoreNew, StoreSetProto};
+use substreams_entity_change::pb::entity::EntityChanges;
+use substreams::store::StoreSet;
+use hex;    
+use substreams_ethereum::pb::eth::v2::BigInt;
+
+
```
Comment (realchoubey) [inline network_substream/src/lib.rs]: Too many extra lines.

File: network_substream/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        .map(|trx| process_transaction_trace(trx, &blk))
+        .collect();
+
+    Ok(TransactionList {
+        transaction_details_list,
+    })
+}
+
+fn process_transaction_trace(trx: eth::TransactionTrace, block: &eth::Block) -> Transaction {
+    // let call_details_list: Vec<T> = trx.calls.into_iter().map(|call| {
```
Comment (realchoubey) [inline network_substream/src/lib.rs]: Remove commented code across repo.

**Review by realchoubey — COMMENTED** [review]
- Add new line at the end of all files
- Remove commented code 
- Remove unwanted entities from schema.graphql and message from proto file.

File: network_substream/Cargo.lock
Comment (harshrjjpt) [reply network_substream/Cargo.lock:1]: no, i removed it

File: network_substream/proto/block_meta.proto
Comment (harshrjjpt) [reply network_substream/proto/block_meta.proto:1]: no, removed extra messages

File: network_substream/src/eth.rs
Comment (harshrjjpt) [reply network_substream/src/eth.rs:1]: nope, removed it!

File: network_substream/src/lib.rs (hunk 7 above)
Comment (harshrjjpt) [reply network_substream/src/lib.rs]: done!

**Review by realchoubey — APPROVED** [review]

**Pull request merged**